pkg/resources/grant_ownership.go

```diff
@@ -0,0 +1,668 @@
+			if objectName, ok := any(data.ObjectName).(sdk.AccountObjectIdentifier); ok {
+				onObject["object_name"] = objectName.Name()
+			} else {
+				onObject["object_name"] = data.ObjectName.FullyQualifiedName()
+			}
+
+			if err := d.Set("on", []any{onObject}); err != nil {
+				return nil, err
+			}
+		case OnAllGrantOwnershipKind, OnFutureGrantOwnershipKind:
+			data := id.Data.(*BulkOperationGrantData)
+
+			on := make(map[string]any)
+			onAllOrFuture := make(map[string]any)
+			onAllOrFuture["object_type_plural"] = data.ObjectNamePlural.String()
+			switch data.Kind {
+			case InDatabaseBulkOperationGrantKind:
+				onAllOrFuture["in_database"] = data.Database.Name()
+			case InSchemaBulkOperationGrantKind:
+				onAllOrFuture["in_schema"] = data.Schema.FullyQualifiedName()
+			}
+
+			switch id.Kind {
+			case OnAllGrantOwnershipKind:
+				on["all"] = []any{onAllOrFuture}
+			case OnFutureGrantOwnershipKind:
+				on["future"] = []any{onAllOrFuture}
+			}
+
+			if err := d.Set("on", []any{on}); err != nil {
+				return nil, err
+			}
+		}
+
+		return []*schema.ResourceData{d}, nil
+	}
+}
+
+func CreateGrantOwnership(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+
+	id, err := createGrantOwnershipIdFromSchema(d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	logging.DebugLogger.Printf("[DEBUG] created identifier from schema: %s", id.String())
+
+	grantOn, err := getOwnershipGrantOn(d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	err = client.Grants.GrantOwnership(
+		ctx,
+		grantOn,
+		getOwnershipGrantTo(d),
+		getOwnershipGrantOpts(id),
+	)
+	if err != nil {
+		return diag.Diagnostics{
+			diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "An error occurred during grant ownership",
+				Detail:   fmt.Sprintf("Id: %s\nError: %s", id.String(), err),
+			},
+		}
+	}
+
+	logging.DebugLogger.Printf("[DEBUG] Setting identifier to %s", id.String())
+	d.SetId(id.String())
+
+	return ReadGrantOwnership(ctx, d, meta)
+}
+
+func DeleteGrantOwnership(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+
+	id, err := ParseGrantOwnershipId(d.Id())
+	if err != nil {
+		return diag.Diagnostics{
+			diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "Failed to parse internal identifier",
+				Detail:   fmt.Sprintf("Id: %s\nError: %s", d.Id(), err),
+			},
+		}
+	}
+
+	grantOn, err := getOwnershipGrantOn(d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if grantOn.Future != nil {
+		// TODO (SNOW-1182623): Still waiting for the response on the behavior/SQL syntax we should use here
+	} else {
+		accountRoleName, err := client.ContextFunctions.CurrentRole(ctx)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		err = client.Grants.GrantOwnership( // TODO: Should we always set outbound privileges to COPY in delete operation or set it to the config value?
```

Comment:
this is similar question like the one we had in copyGrants in other resources, right?

Reply:
Kind of, yeah. When someone defines this resource he/she will be thinking about `on create` effects setting COPY or REVOKE accordingly, but the question is: how do we behave when the resource is being removed? Should we transfer the ownership back with the same option as it was transferred in the Create or similarly in the case of the `view` resource it should be always internally set to COPY `on delete`?